packages/strapi-plugin-content-type-builder/utils/Manager.js

```diff
@@ -0,0 +1,244 @@
+const { findIndex, pullAt, range } = require('lodash');
+const { List } = require('immutable');
+
+class Manager {
+  constructor(state, list, keys, index, layout) {
+    this.state = state;
+    this.keys = keys.split('.');
+    this.layout = layout;
+    this.list = list;
+    this.index = index;
+    this.arrayOfEndLineElements = this.getLinesBound();
+    this.attrToRemoveInfos = this.attrToRemoveInfos();
+  }
+
+  /**
+   * Retrieve the bootstrap col index, name and type of a field
+   * @param {Number} index 
+   * @returns {Object}
+   */
+  getAttrInfos(index) {
+    const name = this.getAttrName(index);
+    const appearance = this.layout.getIn([name, 'appearance']);
+    const type = appearance !== '' && appearance !== undefined ? appearance : this.getType(name);
+    const bootstrapCol = this.getBootStrapCol(type);
+
+    const infos = {
+      bootstrapCol,
+      index,
+      name,
+      type,
+    };
+
+    return infos;
+  }
+
+  getColsToAdd(number) {
```

Comment:
I've already read this method before, in another file. You should specify at the top of the file, why we made that choice.

Reply:
Done